index.rst

```diff
@@ -326,33 +326,167 @@ Analysis of unbinned data is clearly needed for pipeline data quality analyses,
+We also discuss implementation details for creating and updating the Visit database tables.
+
+Similarly to the Butler registry, a PostgreSQL database is the natural choice for the Visit database implementation.
+The Visit database should be replicated at the USDF, and the Summit instance should keep data for at least 30 days.
+
+**Visit and detector tables**
+
+The image headers and the Rapid Analysis outputs are the primary sources of information for the visit and detector tables.
+
+Rapid Analysis will have access to the headers during processing and can consolidate the raw image metadata into per-detector, and per-visit datasets along with the per-visit summary statistics from SFP persisted as Butler datasets.
+Scalar metrics (see :ref:`analysis tools overview <analysis_tools_overview>`) can also be persisted as Butler datasets (see `DMTN-203 Tracking Metrics in Butler`_).
+
+The mechanism for creating these tables in a relational database already exists in DM.
+In particular, if the schema for a Butler dataset is defined in the `Science Data Model`_ (SDM) repository, those definitions can be used to create the corresponding tables.
+
+For the Rapid analysis use case, however, a different mechanism than the one used for data release processing to update the database tables is required, given that the information must be readily available after each visit is processed.
+
+Finally, the Visit and detector tables must enable the selection of data subsets used as input for further analysis (see `DMTN-220 Middleware Support for Campaign Definition and Management`_).
+
+For example, to assess the survey depth from single image requirements, `LPM-17 The LSST System Science Requirements Document`_ specifies the following data selection “r band images in good seeing < 0.7 arcsec (FWHM) on photometric nights (r-band sky brightness > 21 mag/arcsec2) close to the zenith (airmass < 1.2) and total exposure time of 30s.”
+
+The corresponding SQL query (using the `ImSim`_ schema as a reference) would translate to:
+
+.. code:: sql
+
+  SELECT visit
+  FROM Visit v
+  INNER JOIN ccdVisit c ON c.visitId=v.visit
+  WHERE v.band='r'
+  AND c.seeing < 0.7
+  AND c.skyBg > 21
+  AND v.airmass < 1.2
+  AND v.expTime = 30;
+
+
+**Telemetry summary tables**
+
+The current approach for combining visit information with telemetry data is a "join on time," which essentially involves querying two databases, the Butler registry to retrieve the visit's start and end times and the EFD to retrieve the telemetry values in that time range.
+
+The database `use cases`_ confluent page indicates the telemetry information that would be most useful in the Visit database.
+Typically the full time series from the EFD is not required and should consolidate the information (usually spread across multiple EFD topics) into fewer tables to make it easier to query and join with other tables in the Visit database.
```

Comment:
It's not just that it's consolidated, it's that we want to resample it onto visit boundaries via summary statistics of some sort

Reply:
Agreed and expanded that:

```
The database `use cases`_ Confluence page indicates the telemetry information that would be most useful in the Visit database.
Typically the full time series from the EFD is not required and should be resampled onto the visit time range via summary statistics. 
The information, usually spread across multiple EFD topics, should also be consolidated into fewer tables to make it easier to query and join with other tables in the Visit database.
```